rfc/account-usage-get-capability.md

```diff
@@ -0,0 +1,221 @@
+> - If a user didn’t delegate control to us and is using a device without the appropriate delegation from a space. 
+> - If a user logs in to a new agent (B) but creates a new space (2) on the old agent (A), agent B wouldn’t know about space 2.</sub>
+
+
+## Proposal
+
+A new capability that can be invoked in the context of an Account DID (aggregating usage across all spaces). 
+
+### `account-usage/get`
+
+#### Invocation
+
+```ipldsch
+type AccountUsageGet struct {
+  with AccountDID
+  nb optional AccountUsageGetNB
+}
+
+type AccountUsageGetNB struct {
+  space optional [SpaceDID]
+
+  # Optional time period (Unix timestamps in seconds).
+  # If omitted, provider MUST return a current snapshot.
+  period optional Period
+}
+
+type Period struct {
+  from Int # inclusive
+  to Int # inclusive
+}
+```
+
+> example: getting the total usage
+
+```json
+{
+  "iss": "did:key:z6MktfnQz8Kcz5nsC65oyXWFXhbbAZQavjg6LYuHOOOagent",
+  "aud": "did:web:storacha.network",
+  "att": [
+    {
+      "with": "did:mailto:web.mail:alice",
+      "can": "account-usage/get"
+    }
+  ],
+  "prf": [
+    { "/": "bafyAccountDelegationCid" }, 
+    { "/": "bafySessionAttestationCid" }
+  ],
+  "sig": "..."
+}
+````
+
+> example: filtering a single space and period
+
+```json
+{
+  "iss": "did:key:z6MktfnQz8Kcz5nsC65oyXWFXhbbAZQavjg6LYuHOOOagent",
+  "aud": "did:web:storacha.network",
+  "att": [
+    {
+      "with": "did:mailto:web.mail:alice",
+      "can": "account-usage/get",
+      "nb": {
+          "spaces": ["did:key:z6MkuxVKbEvYzXw89c9ESd3xoZ988MFrCgqT5JF5wtBvuYWe"],
+          "period": {
+            "to": 1758111728,
+            "from": 1754006400
+          },
+      }
+    }
+  ],
+ "prf": [
+    { "/": "bafyAccountDelegationCid" }, 
+    { "/": "bafySessionAttestationCid" }
+  ],
+  "sig": "..."
+}
+```
+
+
+
+#### Receipt
+
+```ipldsch
+
+type AccountUsageGetReceipt = {
+  ran: Link<AccountUsageGet>
+  out: Result<AccountUsageGetOk, AccountUsageGetError>
+}
+
+type AccountUsageGetError {
+    message: string
+}
+
+type AccountUsageGetOk {
+    total        Int
+    spaces       {String: SpaceUsage}   # key: SpaceDID
+}
+
+type SpaceUsage {
+  total     Int
+  providers {String: ProviderUsage}   # key: ProviderDID
+}
+
+type ProviderUsage {
+  provider ProviderDID
+  space    SpaceDID
+  period   PeriodISO
+  size SizeDelta
+  events optional [UsageEvent]
+}
+
+type SizeDelta {
+  initial Int
+  final   Int
+}
+
+type UsageEvent {
+  cause     Link
+  delta     Int
+  receiptAt ISO8601Date
+}
+
+type PeriodISO {
+  from ISO8601Date
+  to   ISO8601Date
+}
+
+type ISO8601Date = string
+type ProviderDID = string
+type DID = string
+
+```
+
+> example:
+
+```json
+{
+  "total": 5356848797,
+  "spaces": {
+    "did:key:z6MkuxVKbEvYzXw89c9ESd3xoZ988MFrCgqT5JF5wtBvuYWe": {
+      "total": 5356848797,
+      "providers": {
+        "did:web:web3.storage": {
+          "provider": "did:web:web3.storage",
```

Comment:
Maybe just a list instead of a provider map? Provider is an item for each space anyway.

Alternatively `{ [spaceDID]: Usage }`?